test/scientist/experiment_test.rb

```diff
@@ -262,6 +262,19 @@ def @ex.enabled?
     assert_equal "kaboom", exception.message
   end
 
+  describe "#raise_with" do
+    it "raises custom error if provided" do
+      class CustomError < Scientist::Experiment::MismatchError; end
```

Comment:
It looks like [elsewhere](https://github.com/github/scientist/blob/bdf964f0d8a57b9e8d699d872d30bd0100805dfa/test/scientist/observation_test.rb#L83) in the test setup, we use `Class.new` to create new classes. What do you think of doing the same here with something like:

```ruby
CustomError = Class.new(Scientist::Experiment::MismatchError)
```